src/v/kafka/client/consumer.cc

```diff
@@ -134,15 +135,59 @@ ss::future<> consumer::stop() {
 
     _as.request_abort();
     _on_stopped(_name);
-    return _coordinator->stop()
-      .then([this]() { return _gate.close(); })
+    auto f = ss::now();
+    if (_coordinator) {
+        f = _coordinator->stop();
+    }
+    return f.then([this]() { return _gate.close(); })
       .finally([me{shared_from_this()}] {});
 }
 
 ss::future<> consumer::initialize() {
     vlog(kclog.info, "Consumer: {}: initialize", *this);
     refresh_inactivity_timer();
-    return join();
+    return find_coordinator().then([this](shared_broker_t c) {
```

Comment:
In either case my question is still valid for here then right? If mitigation fails the caller must expect an exception.

Reply:
An exceptional future, yes, but I think that's always been the case.